src/fides/api/ops/service/connectors/base_connector.py

```diff
@@ -112,3 +112,23 @@ def dry_run_query(self, node: TraversalNode) -> Optional[str]:
     @abstractmethod
     def close(self) -> None:
         """Close any held resources"""
+
+
+class LimitedConnector(Generic[DB_CONNECTOR_TYPE], ABC):
+    """Abstract Connector that operates at the Dataset Level, not the Collection level.
+
+    Only supports a subset of functionality that the Connector class supports
+    """
+
+    def __init__(self, configuration: ConnectionConfig):
+        self.configuration = configuration
+        self.hide_parameters = not CONFIG.dev_mode
+        self.db_client: Optional[DB_CONNECTOR_TYPE] = None
+
+    @abstractmethod
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """Used to make a trivial query with the client to ensure secrets are correct.
+
+        If no issues are encountered, this should run without error, otherwise a ConnectionException
+        will be raised.
```

Comment:
The new sovrn consent connector is an example of a connector that does not need a dataset at all, and doesn't need the overhead of the `retrieve_data`, `mask_data` methods that are intended to work on a specific collection or endpoint.